conan/tools/gnu/autotoolstoolchain.py

```diff
@@ -152,14 +152,19 @@ def environment(self):
                             + apple_flags + extra_flags["ldflags"])
         self.defines.extend([self.ndebug, self.gcc_cxx11_abi] + extra_flags["defines"])
 
+        self.cxxflags = self._filter_list_empty_fields(self.cxxflags)
```

Comment:
I think the problem is that the code is not well formed. Probably the ``cxxflags`` should be ready and filtered None values, before even calling ``environment()``, and that function should only convert to an ``Environment`` object. So something like this is possible:

```python
at = AutotoolsToolchain(self)
at.cxxflags = .... # do something
env = at.environmet()
```

Reply:
agree.

Reply:
I will provide the refactor.

Reply:
Now I see why it is done like this. The `cxxflags` are composed by public values that also could be adjusted after instancing the `AutotoolsToolchain`:

```python

at = AutotoolsToolchain(self)
self.libcxx = "whatever" # This affects cxxflags so cannot be computed at constructor.
at.cxxflags = .... # do something
at.environment()
# Now cxxflags is calculated
```

Reply:
Then we probably need to split, have some ``self.extra_cxxflags`` or something, that will be just added to the ``libcxx`` ones, and then not making available the aggregated ``self.cxxflags``? Or leave the ``self.cxxflags`` public, for users, but internally use them only for aggregated to the others?